src/main/java/com/indeed/jiraactions/jiraissues/JiraIssuesProcess.java

```diff
@@ -0,0 +1,145 @@
+    private final List<String> nonApiStatuses = new ArrayList<>(); // Old statuses that don't show up in the API.
+    private final List<String> newFields = new ArrayList<>(); // New fields in updatedIssues not in previous TSV.
+    private List<String[]> newIssues;
+    private List<String> fields; // Fields from updatedIssues.
+    private List<String> oldFields; // Fields from previous TSV
+
+    public JiraIssuesProcess() {}
+
+    public void convertToMap() {
+        for(int issue = 1; issue < newIssues.size(); issue++) {
+            final Map<String, String> mappedLine = new LinkedHashMap<>(fields.size());
+            final String[] line = newIssues.get(issue);
+            if(line == null) {
+                break;
+            } else {
+                for(int i = 0; i < line.length; i++) {
+                    mappedLine.put(fields.get(i), line[i]);
+                }
+                newIssuesMapped.add(mappedLine);
+            }
+        }
+    }
+
+    public void checkAndAddNewFields(){
+        if(oldFields.size() != fields.size()) {
+            for(int i = 0; i < fields.size(); i++) {
+                if(!oldFields.contains(fields.get(i))) {
+                    newFields.add(fields.get(i));
+                }
+            }
+            log.debug("New fields: {}", String.join(" ", newFields));
+        } else {
+            log.info("No new fields.");
+        }
+    }
+
+    /* If the issue is updated through jiraactions it will replace it because that version is the latest.
+     * If it isn't replaced then it gets updated -- only fields involving time are updated so this is really easy.
+     * Issues from jiraactions are removed when they get replaced meaning that the ones remaining are new issues and are therefore added.
+     */
+    public Map<String, String> compareAndUpdate(final String[] issue) {
+        final Map<String, String> mappedLine = new LinkedHashMap<>();
+        // Changes the issue from a String[] to a Map<String, String>
+        for(int i = 0; i < issue.length; i++) {
+            mappedLine.put(oldFields.get(i), issue[i]);
+        }
+
+        for (Map<String, String> updatedIssue: newIssuesMapped) {
+            if (mappedLine.get("issuekey").equals(updatedIssue.get("issuekey"))) {
+                newIssuesMapped.remove(updatedIssue);
+                return updatedIssue;  // Replace
+            }
+        }
+            return updateIssue(mappedLine);   // Update
+    }
+
+    public List<Map<String, String>> getRemainingIssues() {
+        final List<Map<String, String>> addedIssues = new ArrayList<>();
+        if (!newIssues.isEmpty()) {
+            addedIssues.addAll(newIssuesMapped);
+            log.debug("Added {} new issues.", addedIssues.size());
+        }
+        return addedIssues;
+    }
+
+    public Map<String, String> updateIssue(final Map<String, String> mappedLine) {
+        final long DAY = TimeUnit.DAYS.toSeconds(1);
```

Comment:
As I noted before, not every day has this many seconds. We want something that's actually calendar aware here.